server/routes/kanban.js

```diff
@@ -9,52 +10,96 @@ function isAuthenticated(req, res, next) {
+        Card.where('id', newCard.id).fetch({
+          columns: ['id', 'title', 'body', 'priority_id', 'status_id', 'created_by', 'assigned_to'],
+          withRelated: [{
+            'assignedUser': function (qb) {
+              qb.column('id', 'first_name', 'last_name');
+            },
+            'createdByUser': function (y) {
+              y.column('id', 'first_name', 'last_name');
+            }
+          }]
+        })
+          .then((card) => {
+            res.json(card);
+          })
       })
       .catch(function (err) {
         res.status(500).json({ success: false, error: err });
       });
   })
 
-router.route('/:id')
-  .get(function (req, res) { //select card
-    Card.where('id', req.params.id).fetch()
-      .then(function (card) {
-        res.json(card.attributes);
-      })
-      .catch(function (err) {
-        res.status(404).json({ success: false, error: 'card does not exist' });
+router.route('/cards')
+  .get(function (req, res) {
+    Card.fetchAll({
+      columns: ['id', 'title', 'body', 'priority_id', 'status_id', 'created_by', 'assigned_to'],
+      withRelated: [{
+        'assignedUser': function (qb) {
+          qb.column('id', 'first_name', 'last_name');
+        },
+        'createdByUser': function (y) {
+          y.column('id', 'first_name', 'last_name');
+        }
+      }]
+    })
+      .then(function (cards) {
+        res.json({ 'cards': cards });
       });
   })
-  .put(isAuthenticated, function (req, res) { //update card
+
+router.route('/users')
+  .get(function (req, res) {
+    User.fetchAll({ columns: ['id', 'first_name', 'last_name'] })
+      .then(function (users) {
+        res.json({ 'users': users });
+      });
+  });
+
+router.route('/:id')
+  .put(function (req, res) { //update card
     Card.where('id', req.params.id).save({
-      id: req.params.id,
       title: req.body.title,
       body: req.body.body,
       priority_id: req.body.priority_id,
       status_id: req.body.status_id,
-      created_by: req.user.id,
+      created_by: req.created_by,
       assigned_to: req.body.assigned_to
     }, { patch: true })
-      .then(function (err) {
-        res.json({ success: true });
+      .then(function () {
+        Card.where('id', req.params.id).fetch({
+          columns: ['id', 'title', 'body', 'priority_id', 'status_id', 'created_by', 'assigned_to'],
+          withRelated: [{
+            'assignedUser': function (qb) {
+              qb.column('id', 'first_name', 'last_name');
+            },
+            'createdByUser': function (y) {
+              y.column('id', 'first_name', 'last_name');
+            }
+          }]
+        })
+          .then((card) => {
+            res.json(card);
+          })
       })
       .catch(function (err) {
         res.status(500).json({ success: false, error: err });
       });
   })
-  .delete(isAuthenticated, function (req, res) { //delete card
+  .delete(function (req, res) { //delete card
```

Comment:
Why is the isAuthenticated removed here?

Reply:
I have not added authentication to the react page. it works on the server, but to allow the page to add, edit and delete cards, isAuthenticated was temporarily removed.